features/include/pcl/features/good.h

```diff
@@ -0,0 +1,385 @@
+
+      /** \brief compute mean of a given histogram
+	* \param[in] histogram normalized histogram of projection.
+	* \param[out] mean the resultant mean.
+        */        
+      void
+      meanOfHistogram (std::vector<float> histogram, float &mean);
+
+      /** \brief compute variance of a given histogram
+	* \param[in] histogram normalized histogram of projection.
+	* \param[in] mean mean of the given histogram.
+	* \param[out] mean the resultant mean.
+        */        
+      void
+      varianceOfHistogram (std::vector<float> histogram, float mean, float &variance);
+
+      /** \brief concatinating three orthographic projections
+       * three histograms are obtained for the projections; afterwards, two statistic features including entropy and variance have been calculated for each distribution vector;
+       * the histograms are consequently concatenated together using entropy and variance features, to form a single description for the given object. The ordering of the three
+       * histograms is first by decreasing values of entropy. Afterwards the second and third vectors are sorted again by increasing values of variance.
+	* \param[in] maximum_entropy_index index of orthographic projection that has maximum entropy.		
+	* \param[in] normalized_projected_views a vector of vector of float contains three normalized histogram of projected views
+	* \param[out] sorted_normalized_projected_views a vector of float representing the GOOD description for a set of points given by setInputCloud() 
+	* \param[out] name_of_sorted_projected_plane an string represents the order of concatenating projections.
+        */              
+      void
+      objectViewHistogram (int maximum_entropy_index, std::vector<std::vector<float> > normalized_projected_views,
+		      std::vector<float> &sorted_normalized_projected_views,
+		      std::string &name_of_sorted_projected_plane /*debug*/);
+
+      /** \brief compute largest side of boundingbox
+	* \param[in] dimensions shows the dimention of boundingbox
+	* \param[out] largest_side the resultant largest side.
+        */     
+      void
+      computeLargestSideOfBoundingBox (pcl::PointXYZ dimensions, double &largest_side );
+
+      /** \brief compute compute distance between two projections
+	* \param[in] projection1 2D histogram
+	* \param[in] projection2 2D histogram
+	* \param[out] largest_side the resultant distance.
+        */           
+      void
+      computeDistanceBetweenProjections (std::vector <std::vector<float> > projection1, std::vector <std::vector<float> > projection2, float &distance);
+
+      /** \brief initializing a 2D histogram with 0; This function creates a 2D vector with the size of [number_of_bins ? number_of_bins] and initializes all elements with 0 value.
+	* \param[in] number_of_bins number of bin 
+	* \param[out] a_2D_vector
+        */           
+      inline std::vector<std::vector<unsigned int> > initializing2DHistogram (unsigned int number_of_bins);

```

Comment:
This method turns out to be kind of useless because your histogram sizes are know at compile time and  because the actual data initialization is done on the create2DHistogramFromAAAProjection